app/Models/Loggable.php

```diff
@@ -100,27 +129,62 @@ private function determineLogItemType($log)
      */
     public function logCheckin($target, $note)
     {
+        $settings = Setting::getSettings();
         $log = new Actionlog;
         $log->target_type = get_class($target);
         $log->target_id = $target->id;
+
         if (static::class == LicenseSeat::class) {
             $log->item_type = License::class;
             $log->item_id = $this->license_id;
         } else {
             $log->item_type = static::class;
             $log->item_id = $this->id;
         }
+
+
         $log->location_id = null;
         $log->note = $note;
         $log->user_id = Auth::user()->id;
         $log->logaction('checkin from');
 
         $params = [
+            'target' => $target,
             'item' => $log->item,
             'admin' => $log->user,
-            'note' => $note
+            'note' => $note,
+            'target_type' => $log->target_type,
+            'settings' => $settings,
         ];
-        Setting::getSettings()->notify(new CheckinNotification($params));
+
+        // Send to the admin, if settings dictate
+        $recipient = new \App\Models\Recipients\AdminRecipient();
+
+        $checkoutClass = null;
+
+        switch(static::class) {
+            case Asset::class:
+                $checkoutClass = CheckinAssetNotification::class;
+                break;
+            case Accessory::class:
+                $checkoutClass = CheckinAccessoryNotification::class;
+                break;
+            case LicenseSeat::class:
+                $checkoutClass = CheckinLicenseNotification::class;
+                break;
+        }
+
+        if (method_exists($target, 'notify')) {
+            $target->notify(new $checkoutClass($params));
+        }
+
+        if ($settings->admin_cc_email!='') {
```

Comment:
I'd normally agree (in fact you saw me implement it that way) but the problem is that some of those notifiers won't be triggered based on the item category properties, which means the admin will never get then email because the target never does. For example, as an admin, if I say I want emails for every checkout (god help me), checkouts to location won't fire in your scenario (since locations don't have an email address), so the admin will never see that checkout email.